src/commands/Configuration/configure/prefix.ts

```diff
@@ -1,32 +1,111 @@
+import DatabaseManager from "../../../utils/managers/DatabaseManager";
+import { GuildConfiguration } from "../../../utils/types/DatabaseStructures";
 
 const databaseManager: DatabaseManager = new DatabaseManager();
 
-export default class extends SubCommand {
-	constructor(client: BulbBotClient, parent: Command) {
-		super(client, parent, {
-			name: "prefix",
-			clearance: 75,
-			minArgs: 1,
-			maxArgs: -1,
-			argList: ["prefix:String"],
-			usage: "<prefix>",
-			description: "Sets the prefix for the bot.",
-		});
-	}
+async function prefix(interaction: MessageComponentInteraction, client: BulbBotClient) {
+	const config: GuildConfiguration = await databaseManager.getConfig(interaction.guild?.id as Snowflake);
+
+	const placeholderRow = new MessageActionRow().addComponents(
+		new MessageSelectMenu()
+			.setCustomId("placeholder")
+			.setPlaceholder(`Current prefix: ${config.prefix}`)
+			.setDisabled(true)
+			.addOptions([
+				{
+					label: "Placeholder",
+					value: "placeholder",
+				},
+			]),
+	);
+
+	const [back, change, header, reset] = [
+		await client.bulbutils.translate("config_button_back", interaction.guild?.id, {}),
+		await client.bulbutils.translate("config_prefix_button_change", interaction.guild?.id, {}),
+		await client.bulbutils.translate("config_prefix_header", interaction.guild?.id, {}),
+		await client.bulbutils.translate("config_prefix_button_reset", interaction.guild?.id, {}),
+	];
+
+	const buttonRow = new MessageActionRow().addComponents([
+		new MessageButton().setCustomId("back").setLabel(back).setStyle("DANGER"),
+		new MessageButton().setCustomId("change").setLabel(change).setStyle("SUCCESS"),
+		new MessageButton()
+			.setCustomId("reset")
+			.setLabel(reset)
+			.setStyle("PRIMARY")
+			.setDisabled(config.prefix === "!"),
+	]);
+
+	interaction.deferred
+		? await interaction.editReply({ content: header, components: [placeholderRow, buttonRow] })
+		: await interaction.update({ content: header, components: [placeholderRow, buttonRow] });
+
+	const filter = i => i.user.id === interaction.user.id;
+	const collector = interaction.channel?.createMessageComponentCollector({ filter, time: 60000, max: 1 });
 
-	public async run(context: CommandContext, args: string[]): Promise<void | Message> {
-		const prefix = args.join(" ");
+	collector?.on("collect", async (i: MessageComponentInteraction) => {
+		if (i.isButton()) {
+			switch (i.customId) {
+				case "back":
+					await require("./main").default(i, client);
+					break;
+				case "reset":
+					await interaction.followUp({
+						content: await client.bulbutils.translate("config_prefix_reset", interaction.guild?.id, {}),
+						ephemeral: true,
+					});
+					await databaseManager.setPrefix(interaction.guild?.id as Snowflake, "!");
+					await prefix(i, client);
+					break;
+				case "change":
+					buttonRow.components[0].setDisabled(true);
+					buttonRow.components[1].setDisabled(true);
+					buttonRow.components[2].setDisabled(true);
```

Comment:
```suggestion
					for (const c of buttonRow.components) {
						buttonRow.components[c].setDisabled(true);
					}
```
Not really sure if it will work without the `number` type.

Reply:
Same here, for of gives you an object not an index

Reply:
Same as https://github.com/TeamBulbbot/bulbbot/pull/339#discussion_r833321718